src/sentry/integrations/slack/service.py

```diff
@@ -133,7 +133,12 @@ def notify_all_threads_for_activity(self, activity: Activity) -> None:
             )
             return None
 
-        if activity.user_id is None:
+        uptime_resolved_notification = (
+            activity.type == ActivityType.SET_RESOLVED.value
+            and activity.group.issue_category == GroupCategory.UPTIME
+        )
```

Comment:
I have a feeling this is going to be a problem for crons, metric alerts, etc. Probably we should build this into the group type to be more general, so that we don't need to keep adding to this list.

Not needed today but I think we should make this general soon.